static/js/src/advantage/subscribe/react/components/Summary.jsx

```diff
@@ -0,0 +1,139 @@
+import React from "react";
+import { Row, Col } from "@canonical/react-components";
+import { add, format } from "date-fns";
+import { formatter } from "../../renderers/form-renderer";
+import usePreview from "../APICalls/usePreview";
+import useProduct from "../APICalls/useProduct";
+
+const DATE_FORMAT = "dd MMMM yyyy";
+
+function Summary() {
+  const { product, quantity } = useProduct();
+  const { data: preview } = usePreview();
+
+  return (
+    <section
+      id="summary-section"
+      className="p-strip is-shallow u-no-padding--top"
+    >
+      <Row className="u-no-padding u-sv1">
+        <Col size="4">
+          <div className="u-text-light">Plan type:</div>
+        </Col>
+        <Col size="8">
+          <div>{product?.name}</div>
+        </Col>
+      </Row>
+      <Row className="u-no-padding u-sv1">
+        <Col size="4">
+          <div className="u-text-light">Machines:</div>
+        </Col>
+        <Col size="8">
+          <div>
+            {quantity} x {formatter.format(product?.price?.value / 100)}
```

Comment:
If there is an issue when deriving these values then you might end up with some funny output here. Maybe this should be calculated outside of the return so that sane defaults could be returned in the event of an error.

Reply:
This component is conditionally rendered depending on the presence of those values so I think it's pretty safe. Do you think I should still add another layer of safety?
I'm not sure what sane default could be returned for a missing price on a purchase summary. In this case, the absolute worst thing that can happen (And don't think it can) is to display NaN